app/models/goal.py

```diff
@@ -1,5 +1,27 @@
-from sqlalchemy.orm import Mapped, mapped_column
+from sqlalchemy.orm import Mapped, mapped_column, relationship
 from ..db import db
+from typing import List, Optional
+
+from typing import TYPE_CHECKING
+if TYPE_CHECKING: from .task import Task
 
 class Goal(db.Model):
     id: Mapped[int] = mapped_column(primary_key=True, autoincrement=True)
+    title: Mapped[str]
+    tasks: Mapped[List["Task"]] = relationship("Task", back_populates="goal")  
+    
+    @classmethod
+    def from_dict(cls, goal_data):
+        new_goal = Goal(title=goal_data["title"])
+        return new_goal
```

Comment:
Since we have access to the class with the `cls` keyword that's necessary for defining a class method, we can use `cls` instead of `Goal`. 

Using the more generic `cls` means that if we ever update the class name, we won't need to update line 15.
```suggestion
        return cls(title=goal_data["title"])
```